### ISSUE TITLE
Hyper-V Builder - Add support for Secure Boot of non-Windows OS's

### ISSUE THREAD

**Issue by evancox10** [hashicorp/packer#5087]
Packer provides the enable_secure_boot parameter to turn on secure boot, this works if the guest OS is Windows, but if it's another Secure Boot supported OS such as Ubuntu it does not work. The reason why is that the SecureBootTemplate has to be changed from the default one to MicrosoftUEFICertificateAuthority.

I would like to see support for choosing the SecureBootTemplate along with the choice of enabling Secure Boot. Changing the template is easy, and can be done in the same Power Shell command as turning on SB:
```Set-VMFirmware -VMName dtlinux -EnableSecureBoot On -SecureBootTemplate 'MicrosoftUEFICertificateAuthority'```

Good source for more info: http://www.altaro.com/hyper-v/hyper-v-2016-support-linux-secure-boot/

**Comment by taliesins** [comment]
Yes I would like a PR for this. I am assuming that if we would have 1 parameter in Packer. 

If its not a empty it would add `-EnableSecureBoot On`

**Comment by PatrickLang** [comment]
@shirgall Do you have any feedback on this? Do you have a list of which distros work with `MicrosoftUEFICertificateAuthority` and secure boot?

**Comment by shirgall** [comment]
All the major enterprise distributions support secure boot with the latest releases. The odd ones out are Debian and CentOS.

**Comment by synax** [comment]
I actually need this functionality in a current project, so I thought I'll give it a try. I've added an additional property called "secure_boot_template" which can be used to specify the secure boot template. 

First I looked in to changing the "enable_secure_boot" property to a string, which would allow you to just specify the template and if you specify something here, it would enable secure boot implicitly. I think having a second field gives more control. For example, with two options it is possible to disable secure boot and still pre-configure the correct template for a Linux template, in case a user enables it later.

**Comment by uberjew666** [comment]
+1
